it/controllers/expenses/RemoveExpensesControllerISpec.scala

```diff
@@ -0,0 +1,225 @@
+  }
+
+  object ExpectedAgentCY extends SpecificExpectedResults {
+    val expectedTitle = "Are you sure you want to remove your client’s expenses?"
+    val expectedHeading = "Are you sure you want to remove your client’s expenses?"
+  }
+
+  val userScenarios: Seq[UserScenario[CommonExpectedResults, SpecificExpectedResults]] = Seq(
+    UserScenario(isWelsh = false, isAgent = false, CommonExpectedEN, Some(ExpectedIndividualEN)),
+    UserScenario(isWelsh = false, isAgent = true, CommonExpectedEN, Some(ExpectedAgentEN)),
+    UserScenario(isWelsh = true, isAgent = false, CommonExpectedCY, Some(ExpectedIndividualCY)),
+    UserScenario(isWelsh = true, isAgent = true, CommonExpectedCY, Some(ExpectedAgentCY))
+  )
+
+  ".show" should {
+
+    import Selectors._
+
+    userScenarios.foreach { user =>
+
+      val common = user.commonExpectedResults
+      val specific = user.specificExpectedResults.get
+
+      s"language is ${welshTest(user.isWelsh)} and request is from an ${agentTest(user.isAgent)}" should {
+        "render the remove expenses page when user has expenses" which {
+          lazy val result: WSResponse = {
+            dropExpensesDB()
+            authoriseAgentOrIndividual(user.isAgent)
+            userDataStub(IncomeTaxUserData(Some(model)), nino, taxYearEOY)
+            urlGet(fullUrl(removeExpensesUrl(taxYearEOY)), welsh = user.isWelsh, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+          }
+
+          s"has an OK ($OK) status" in {
+            result.status shouldBe OK
+          }
+
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+          welshToggleCheck(user.isWelsh)
+          titleCheck(specific.expectedTitle)
+          h1Check(specific.expectedHeading)
+          captionCheck(common.expectedCaption)
+          textOnPageCheck(common.expectedRemoveExpensesText, paragraphTextSelector)
+          buttonCheck(common.expectedRemoveExpensesButton, removeExpensesButtonSelector)
+          linkCheck(common.expectedCancelLink, cancelLinkSelector, employmentSummaryUrl(taxYearEOY))
+          formPostLinkCheck(removeExpensesUrl(taxYearEOY), formSelector)
+        }
+
+        "redirect to the overview page" when {
+          "it is not end of year" which {
+            lazy val result: WSResponse = {
+              dropExpensesDB()
+              authoriseAgentOrIndividual(user.isAgent)
+              userDataStub(IncomeTaxUserData(Some(modelToDelete)), nino, taxYear)
+              urlGet(fullUrl(removeExpensesUrl(taxYear)), welsh = user.isWelsh, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYear)), follow = false)
+            }
+
+            s"has a SEE_OTHER ($SEE_OTHER) status" in {
+              result.status shouldBe SEE_OTHER
+              result.header("location").contains(overviewUrl(taxYear)) shouldBe true
+            }
+          }
+
+
+        }
+      }
+    }
+  }
+
+  ".submit" should {
+
+    userScenarios.foreach { user =>
```

Comment:
I think the userScenarios.foreach loop can be taken out since you're not checking for content on the page